#! /bin/sh /usr/share/dpatch/dpatch-run
## 25_fix_stereo_line_capture.dpatch by  <jeff@boxybutgood.com>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes stereo line capture bug in simple mixer API: 
## DP: http://hg-mirror.alsa-project.org/alsa-lib?cmd=changeset;node=71eb744cd5
## DP: be42cd7ebd14edf5f3f2e2da6ba35d;style=gitweb

@DPATCH@

diff -ru alsa-lib-1.0.10.orig/src/mixer/simple_none.c alsa-lib-1.0.10/src/mixer/simple_none.c
--- alsa-lib-1.0.10.orig/src/mixer/simple_none.c	2005-08-24 05:41:16.000000000 -0500
+++ alsa-lib-1.0.10/src/mixer/simple_none.c	2005-12-02 07:39:24.000000000 -0600
@@ -768,7 +768,9 @@
 	if (cchannels > 1) {
 		if (simple->ctls[CTL_CAPTURE_ROUTE].elem ||
 		    (simple->ctls[CTL_CAPTURE_SWITCH].elem &&
-		     simple->ctls[CTL_CAPTURE_SWITCH].values > 1)) {
+		     simple->ctls[CTL_CAPTURE_SWITCH].values > 1) ||
+		    (simple->ctls[CTL_CAPTURE_SOURCE].elem &&
+		     simple->ctls[CTL_CAPTURE_SOURCE].values > 1)) {
 			caps &= ~SM_CAP_CSWITCH_JOIN;
 		}
 		if (simple->ctls[CTL_CAPTURE_VOLUME].elem &&
